mediapipe/tasks/ios/test/vision/image_classifier/ImageClassifierTests.swift

```diff
@@ -687,4 +689,125 @@ class ImageClassifierTests: XCTestCase {
+      }
+     expectation.fulfill()
+    }
+  
+    let imageClassifier = try XCTUnwrap(ImageClassifier(options: 
+      imageClassifierOptions))
+
+    let mpImage = try XCTUnwrap(
+      imageWithFileInfo(ImageClassifierTests.burgerImage))
+
+    XCTAssertNoThrow(
+      try imageClassifier.classifyAsync(
+        image: mpImage,
+        timestampMs: 100))
+      
+    XCTAssertThrowsError(
+      try imageClassifier.classifyAsync(
+        image: mpImage,
+        timestampMs: 0)) {(error) in
+          assertEqualErrorDescriptions(
+          error,
+          expectedLocalizedDescription: """
+          INVALID_ARGUMENT: Input timestamp must be monotonically \
+          increasing.
+          """)
+    }
+   
+    wait(for:[expectation], timeout: 0.1)
+  }
+
+  func testClassifyWithLiveStreamModeSucceeds() throws {
+    let imageClassifierOptions =
+    try XCTUnwrap(
+      imageClassifierOptionsWithModelPath(
+        ImageClassifierTests.floatModelPath))
+    
+    imageClassifierOptions.runningMode = .liveStream
+
+    let maxResults = 3
+    imageClassifierOptions.maxResults = maxResults
+
+    let iterationCount = 100;
+ 
+   // Because of flow limiting, we cannot ensure that the callback will be 
+   // invoked `iterationCount` times.
+   // An normal expectation will fail if expectation.fullfill() is not called 
+   // `expectation.expectedFulfillmentCount` times.
+   // If `expectation.isInverted = true`, the test will only succeed if 
+   // expectation is not fullfilled for the specified `expectedFulfillmentCount`.
+   // Since in our case we cannot predict how many times the expectation is 
+   // supposed to be fullfilled setting,
+   // `expectation.expectedFulfillmentCount` = `iterationCount` and 
+   // `expectation.isInverted = true` ensures that test succeeds if 
+   // expectation is not fullfilled `iterationCount` times.
+    let expectation = expectation(description: "liveStreamClassify")
+    expectation.expectedFulfillmentCount = iterationCount;
+    expectation.isInverted = true;
+  
+    imageClassifierOptions.completion = {(
+      result: ImageClassifierResult?, 
+      timestampMs: Int, 
+      error: Error?) -> () in
+      do {
+        try self.assertImageClassifierResult(
+          try XCTUnwrap(result),
+          hasCategoryCount: maxResults,
+          andCategories: 
+            ImageClassifierTests
+              .expectedResultsClassifyBurgerImageWithFloatModel)
+      }
+      catch {
+          // Any errors will be thrown by the wait() method of the expectation.
+      }
+     expectation.fulfill()
+    }
+  
+    let imageClassifier = try XCTUnwrap(ImageClassifier(options: 
+      imageClassifierOptions))
+
+     let mpImage = try XCTUnwrap(
+      imageWithFileInfo(ImageClassifierTests.burgerImage))
+    
+    for i in 0..<iterationCount {
+      XCTAssertNoThrow(
+        try imageClassifier.classifyAsync(
+          image: mpImage,
+          timestampMs: i))
```

Comment:
Can we work around flow limiting by passing a higher gap between timestamps (i.e. `i * 1000`)?

Reply:
I don't think the goal of this test should be to test flow limiting. We should test the classifyAsync call in LiveStream mode with multiple packages. The fact that there is flow limiting is an implementation detail (albeit one that makes the test harder to write). The current Swift test would fail if for some reason flow limiting was turned off or did no kick in, versus the C++ test would still succeed. I think we should try to figure out if we can achieve the same behavior on iOS (and maybe that's just raising the expectation limit to 101).

Reply:
I have set it to 101 for now.